mapows.c

```diff
@@ -175,8 +175,14 @@ static int msOWSPreParseRequest(cgiRequestObj *request,
     }
 
     /* Get service, version and request from root */
-    ows_request->service = (char *) xmlGetProp(root, BAD_CAST "service");
-    ows_request->version = (char *) xmlGetProp(root, BAD_CAST "version");
+    char* tmp = NULL;
```

Comment:
Memory issue here?

Reply:
yes...
Pedantically, xmlGetProp() returns a pointer that should be free'd with xmlFree() (see http://www.xmlsoft.org/html/libxml-tree.html#xmlGetProp), not with msFree(), but both are likely just going through system free(). The issue here is more than you need to call xmlFree(tmp) after setting ->service .

So the correct code is
```
 if ((tmp = (char*)xmlGetProp(root, BAD_CAST "service")) != NULL) {
        ows_request->service = msStrdup(tmp);
        xmlFree(tmp);
   }
    if ((tmp = (char*)xmlGetProp(root, BAD_CAST "version")) != NULL) {
        ows_request->version = msStrdup(tmp);
        xmlFree(tmp);
    }
```

Reply:
Thanks @rouault - somehow I missed this comment, but (eventually!) came to the same solution. 
I initially copied the approach from: https://github.com/MapServer/MapServer/blob/2f74ad0bb9c26793ea08643576cbbd46aa11cbb6/mapwcs.cpp#L371

I don't see why it works there (there seems to be no `xmlFree(tmp)` in this function), yet the tests pass with no memory leaks.